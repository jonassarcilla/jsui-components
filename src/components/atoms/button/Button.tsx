// import '@/index.css';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';
import * as React from 'react';

import cn from '@/lib/utils';

const buttonVariants = cva(
  'cv-inline-flex cv-items-center cv-justify-center cv-whitespace-nowrap cv-rounded-md cv-text-sm cv-font-medium cv-ring-offset-background cv-transition-colors focus-visible:cv-outline-none focus-visible:cv-ring-2 focus-visible:cv-ring-ring focus-visible:cv-ring-offset-2 disabled:cv-pointer-events-none disabled:cv-opacity-50',
  {
    variants: {
      variant: {
        default:
          'cv-bg-primary cv-text-primary-foreground hover:cv-bg-primary/90',
        destructive:
          'cv-bg-destructive cv-text-destructive-foreground hover:cv-bg-destructive/90',
        outline:
          'cv-border cv-border-input cv-bg-background hover:cv-bg-accent hover:cv-text-accent-foreground',
        secondary:
          'cv-bg-secondary cv-text-secondary-foreground hover:cv-bg-secondary/80',
        ghost: 'hover:cv-bg-accent hover:cv-text-accent-foreground',
        link: 'cv-text-primary cv-underline-offset-4 hover:cv-underline',
      },
      size: {
        default: 'cv-h-10 cv-px-4 cv-py-2',
        sm: 'cv-h-9 cv-rounded-md cv-px-3',
        lg: 'cv-h-11 cv-rounded-md cv-px-8',
        icon: 'cv-h-10 cv-w-10',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
    },
  }
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  id: string;
  asChild?: boolean;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ id, className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : 'button';
    return (
      <Comp
        data-testid={id}
        className={cn(buttonVariants({ variant, size, className }))}
        ref={ref}
        {...props}
      />
    );
  }
);
Button.displayName = 'Button';

export { Button, buttonVariants };
