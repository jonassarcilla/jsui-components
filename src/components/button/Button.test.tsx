import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Button from './Button';

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button>Click Me</Button>);

    expect(container.firstChild).toHaveClass('cv-bg-violet-500');
    expect(container.firstChild).toHaveClass('cv-text-white');
    expect(container.firstChild).toHaveClass('cv-rounded-md');
    expect(container.firstChild).toHaveClass('cv-px-4');
  });

  it('renders correctly with left icon', () => {
    const { container } = render(
      <Button leftIcon={<span>ğŸ‘ˆ</span>}>Click Me</Button>
    );

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('ğŸ‘ˆ');
  });

  it('renders correctly with right icon', () => {
    const { container } = render(
      <Button rightIcon={<span>ğŸ‘‰</span>}>Click Me</Button>
    );

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('renders correctly with both icons', () => {
    const { container } = render(
      <Button leftIcon={<span>ğŸ‘ˆ</span>} rightIcon={<span>ğŸ‘‰</span>} />
    );

    expect(container).toHaveTextContent('ğŸ‘ˆ');
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('renders correctly with label', () => {
    const { container } = render(<Button label="Click Me" />);

    expect(container).toHaveTextContent('Click Me');
  });

  it('renders correctly with label and left icon', () => {
    const { container } = render(
      <Button
        label="Click Me"
        leftIcon={<span>ğŸ‘ˆ</span>}
        rightIcon={<span>ğŸ‘‰</span>}
      />
    );

    expect(container).toHaveTextContent('Click Me');
    expect(container).toHaveTextContent('ğŸ‘ˆ');
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('render correctly with size prop', () => {
    const { container } = render(<Button size="default" />);

    const { container: containerSmall } = render(<Button size="small" />);

    const { container: containerLarge } = render(<Button size="large" />);

    const { container: containerXXL } = render(<Button size="xxl" />);

    expect(container.firstChild).toHaveClass('cv-text-base');
    expect(containerSmall.firstChild).toHaveClass('cv-text-sm');
    expect(containerLarge.firstChild).toHaveClass('cv-text-lg');
    expect(containerXXL.firstChild).toHaveClass('cv-text-2xl');
  });

  it('render correctly with rounded prop', () => {
    const { container } = render(<Button rounded="full" />);

    const { container: containerLg } = render(<Button rounded="lg" />);

    const { container: containerNone } = render(<Button rounded="none" />);

    const { container: containerXL } = render(<Button rounded="xl" />);

    const { container: containerXXL } = render(<Button rounded="xxl" />);

    const { container: containerDefault } = render(
      <Button rounded="default" />
    );

    const { container: containerSM } = render(<Button rounded="sm" />);

    expect(container).toMatchSnapshot();
    expect(containerLg).toMatchSnapshot();
    expect(containerNone).toMatchSnapshot();
    expect(containerXL).toMatchSnapshot();
    expect(containerXXL).toMatchSnapshot();
    expect(containerDefault).toMatchSnapshot();
    expect(containerSM).toMatchSnapshot();

    expect(container.firstChild).toHaveClass('cv-rounded-full');
    expect(containerLg.firstChild).toHaveClass('cv-rounded-lg');
    expect(containerNone.firstChild).toHaveClass('cv-rounded-none');
    expect(containerXL.firstChild).toHaveClass('cv-rounded-xl');
    expect(containerXXL.firstChild).toHaveClass('cv-rounded-2xl');
    expect(containerDefault.firstChild).toHaveClass('cv-rounded-md');
    expect(containerSM.firstChild).toHaveClass('cv-rounded-sm');
  });

  it('render correctly with buttonType prop', () => {
    const { container: containerPrimary } = render(
      <Button buttonType="primary" />
    );

    const { container } = render(<Button buttonType="secondary" />);

    const { container: containerWarning } = render(
      <Button buttonType="warning" />
    );

    const { container: containerOutline } = render(
      <Button buttonType="outline" />
    );

    const { container: containerDisabled } = render(
      <Button buttonType="disabled" />
    );

    const { container: containerError } = render(<Button buttonType="error" />);

    expect(container).toMatchSnapshot();
    expect(containerPrimary).toMatchSnapshot();
    expect(containerWarning).toMatchSnapshot();
    expect(containerOutline).toMatchSnapshot();
    expect(containerDisabled).toMatchSnapshot();
    expect(containerError).toMatchSnapshot();

    expect(containerPrimary.firstChild).toHaveClass(
      'cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600'
    );
    expect(container.firstChild).toHaveClass(
      'cv-bg-gray-200 cv-text-gray-600 cv-border-gray-200 hover:cv-bg-gray-300'
    );
    expect(containerOutline.firstChild).toHaveClass(
      'cv-bg-white cv-text-gray-600 hover:cv-bg-gray-100 cv-border hover:cv-border-gray-100 cv-border-gray-300 hover:cv-shadow-md'
    );
    expect(containerWarning.firstChild).toHaveClass(
      'cv-bg-yellow-500 cv-text-white cv-border-yellow-500 hover:cv-bg-yellow-600'
    );

    expect(containerDisabled.firstChild).toHaveClass(
      'cv-bg-black cv-text-white cv-border-black cv-cursor-not-allowed'
    );

    expect(containerError.firstChild).toHaveClass(
      'cv-bg-red-500 cv-text-white cv-border-red-500 hover:cv-bg-red-600'
    );
  });
});

describe('Button - 1', () => {
  it('renders correctly', () => {
    const { container } = render(<Button>Click Me</Button>);

    expect(container.firstChild).toHaveClass('cv-bg-violet-500');
    expect(container.firstChild).toHaveClass('cv-text-white');
    expect(container.firstChild).toHaveClass('cv-rounded-md');
    expect(container.firstChild).toHaveClass('cv-px-4');
  });

  it('renders correctly with left icon', () => {
    const { container } = render(
      <Button leftIcon={<span>ğŸ‘ˆ</span>}>Click Me</Button>
    );

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('ğŸ‘ˆ');
  });

  it('renders correctly with right icon', () => {
    const { container } = render(
      <Button rightIcon={<span>ğŸ‘‰</span>}>Click Me</Button>
    );

    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('renders correctly with both icons', () => {
    const { container } = render(
      <Button leftIcon={<span>ğŸ‘ˆ</span>} rightIcon={<span>ğŸ‘‰</span>} />
    );

    expect(container).toHaveTextContent('ğŸ‘ˆ');
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('renders correctly with label', () => {
    const { container } = render(<Button label="Click Me" />);

    expect(container).toHaveTextContent('Click Me');
  });

  it('renders correctly with label and left icon', () => {
    const { container } = render(
      <Button
        label="Click Me"
        leftIcon={<span>ğŸ‘ˆ</span>}
        rightIcon={<span>ğŸ‘‰</span>}
      />
    );

    expect(container).toHaveTextContent('Click Me');
    expect(container).toHaveTextContent('ğŸ‘ˆ');
    expect(container).toHaveTextContent('ğŸ‘‰');
  });

  it('render correctly with size prop', () => {
    const { container } = render(<Button size="default" />);

    const { container: containerSmall } = render(<Button size="small" />);

    const { container: containerLarge } = render(<Button size="large" />);

    const { container: containerXXL } = render(<Button size="xxl" />);

    expect(container.firstChild).toHaveClass('cv-text-base');
    expect(containerSmall.firstChild).toHaveClass('cv-text-sm');
    expect(containerLarge.firstChild).toHaveClass('cv-text-lg');
    expect(containerXXL.firstChild).toHaveClass('cv-text-2xl');
  });

  it('render correctly with rounded prop', () => {
    const { container } = render(<Button rounded="full" />);

    const { container: containerLg } = render(<Button rounded="lg" />);

    const { container: containerNone } = render(<Button rounded="none" />);

    const { container: containerXL } = render(<Button rounded="xl" />);

    const { container: containerXXL } = render(<Button rounded="xxl" />);

    const { container: containerDefault } = render(
      <Button rounded="default" />
    );

    const { container: containerSM } = render(<Button rounded="sm" />);

    expect(container).toMatchSnapshot();
    expect(containerLg).toMatchSnapshot();
    expect(containerNone).toMatchSnapshot();
    expect(containerXL).toMatchSnapshot();
    expect(containerXXL).toMatchSnapshot();
    expect(containerDefault).toMatchSnapshot();
    expect(containerSM).toMatchSnapshot();

    expect(container.firstChild).toHaveClass('cv-rounded-full');
    expect(containerLg.firstChild).toHaveClass('cv-rounded-lg');
    expect(containerNone.firstChild).toHaveClass('cv-rounded-none');
    expect(containerXL.firstChild).toHaveClass('cv-rounded-xl');
    expect(containerXXL.firstChild).toHaveClass('cv-rounded-2xl');
    expect(containerDefault.firstChild).toHaveClass('cv-rounded-md');
    expect(containerSM.firstChild).toHaveClass('cv-rounded-sm');
  });

  it('render correctly with buttonType prop', () => {
    const { container: containerPrimary } = render(
      <Button buttonType="primary" />
    );

    const { container } = render(<Button buttonType="secondary" />);

    const { container: containerWarning } = render(
      <Button buttonType="warning" />
    );

    const { container: containerOutline } = render(
      <Button buttonType="outline" />
    );

    const { container: containerDisabled } = render(
      <Button buttonType="disabled" />
    );

    const { container: containerError } = render(<Button buttonType="error" />);

    expect(container).toMatchSnapshot();
    expect(containerPrimary).toMatchSnapshot();
    expect(containerWarning).toMatchSnapshot();
    expect(containerOutline).toMatchSnapshot();
    expect(containerDisabled).toMatchSnapshot();
    expect(containerError).toMatchSnapshot();

    expect(containerPrimary.firstChild).toHaveClass(
      'cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600'
    );
    expect(container.firstChild).toHaveClass(
      'cv-bg-gray-200 cv-text-gray-600 cv-border-gray-200 hover:cv-bg-gray-300'
    );
    expect(containerOutline.firstChild).toHaveClass(
      'cv-bg-white cv-text-gray-600 hover:cv-bg-gray-100 cv-border hover:cv-border-gray-100 cv-border-gray-300 hover:cv-shadow-md'
    );
    expect(containerWarning.firstChild).toHaveClass(
      'cv-bg-yellow-500 cv-text-white cv-border-yellow-500 hover:cv-bg-yellow-600'
    );

    expect(containerDisabled.firstChild).toHaveClass(
      'cv-bg-black cv-text-white cv-border-black cv-cursor-not-allowed'
    );

    expect(containerError.firstChild).toHaveClass(
      'cv-bg-red-500 cv-text-white cv-border-red-500 hover:cv-bg-red-600'
    );
  });
});
