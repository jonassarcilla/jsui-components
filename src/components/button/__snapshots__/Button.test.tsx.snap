// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > render correctly with buttonType prop 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-gray-200 cv-text-gray-600 cv-border-gray-200 hover:cv-bg-gray-300 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with buttonType prop 2`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with buttonType prop 3`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-yellow-500 cv-text-white cv-border-yellow-500 hover:cv-bg-yellow-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with buttonType prop 4`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-white cv-text-gray-600 hover:cv-bg-gray-100 cv-border hover:cv-border-gray-100 cv-border-gray-300 hover:cv-shadow-md cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with buttonType prop 5`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-black cv-text-white cv-border-black cv-cursor-not-allowed cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with buttonType prop 6`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-red-500 cv-text-white cv-border-red-500 hover:cv-bg-red-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-full"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 2`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-lg"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 3`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-none"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 4`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-xl"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 5`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-2xl"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 6`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button > render correctly with rounded prop 7`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-sm"
    type="button"
  />
</div>
`;

exports[`Button > renders correctly with left icon 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  >
    <span>
      ðŸ‘ˆ
    </span>
  </button>
</div>
`;

exports[`Button > renders correctly with right icon 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  >
    <span>
      ðŸ‘‰
    </span>
  </button>
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-gray-200 cv-text-gray-600 cv-border-gray-200 hover:cv-bg-gray-300 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 2`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 3`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-yellow-500 cv-text-white cv-border-yellow-500 hover:cv-bg-yellow-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 4`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-white cv-text-gray-600 hover:cv-bg-gray-100 cv-border hover:cv-border-gray-100 cv-border-gray-300 hover:cv-shadow-md cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 5`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-black cv-text-white cv-border-black cv-cursor-not-allowed cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with buttonType prop 6`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-red-500 cv-text-white cv-border-red-500 hover:cv-bg-red-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-full"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 2`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-lg"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 3`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-none"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 4`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-xl"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 5`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-2xl"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 6`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  />
</div>
`;

exports[`Button - 1 > render correctly with rounded prop 7`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-sm"
    type="button"
  />
</div>
`;

exports[`Button - 1 > renders correctly with left icon 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  >
    <span>
      ðŸ‘ˆ
    </span>
  </button>
</div>
`;

exports[`Button - 1 > renders correctly with right icon 1`] = `
<div>
  <button
    class="flex flex-row gap-x-4 disabled:cursor-not-allowed items-center justify-center cv-bg-violet-500 cv-text-white cv-border-violet-500 hover:cv-bg-violet-600 cv-text-base cv-py-2 cv-px-4 cv-rounded-md"
    type="button"
  >
    <span>
      ðŸ‘‰
    </span>
  </button>
</div>
`;
